Added Additional Base Functionality

inspectorG4dget opened the pull request:
Added functionality:
All prints now show the filename, function, line number, and description

- [x] I have visited the [source website], and in particular
  read the [known issues]
- [x] I have searched through the [issue tracker] for duplicates
- [x] I have mentioned version numbers, operating system and
  environment, where applicable:
  ```python
  import tqdm, sys
  print(tqdm.__version__, sys.version, sys.platform)
  ```
- [x] If applicable, I have mentioned the relevant/related issue(s)

  [source website]: https://github.com/tqdm/tqdm/
  [known issues]: https://github.com/tqdm/tqdm/#faq-and-known-issues
  [issue tracker]: https://github.com/tqdm/tqdm/issues?q=

inspectorG4dget committed:
Added Additional Base Functionality

Added functionality:
All prints now show the filename, function, line number, and description

inspectorG4dget committed:
Added current time

TQDM now outputs the print time as well

inspectorG4dget committed:
Printing also conforms to bar format

Bar format has been ported over to printing

casperdcl commented:
Maybe nice to have as an example of extending `tqdm` but don't really see this in core. Could you move this to the `examples/` folder?

inspectorG4dget commented:
In fact, I've added my extensions to patch.py by subclassing the main class, so that it does not conflict with mainline development/API. The proper way to use this patch would therefore be `from tqdm.patch import tqdm, print`. 

In any case, I will add some usage examples to show how this works

inspectorG4dget committed:
Hardening the last commit

This is what the last commit should have been.
Robustified computing the numebr columns across *NIX and PC (by using a default for PC)
Removed commented out debug code
Added print functionality
Some basic refactoring for future maintenance

inspectorG4dget committed:
Fixes #1,2
